Project finished

Eyad-Mostafa opened the pull request:
It's my first app using SQL server, LINQ.
I hope that it's good :D
I have finished all requirements (as I think).
I haven't finished the challenge, I need to better practice SQL but I will back to this challenge someday.

Eyad-Mostafa commented:
Sorry, but I really don't know what is the Issue in the README file

Eyad-Mostafa commented:
Hey @chrisjamiecarter 
Sorry for this mistake. 
I think it works now.
I have deleted the whole folder and made a new one and added the project folders one by one carefully :D

Eyad-Mostafa commented:
There was a mistakenly added folder 🫤, sorry for this

chrisjamiecarter commented on Flashcards.Eyad-Mostafa/FlashCards/Database/DatabaseManager.cs:
```diff
@@ -0,0 +1,341 @@
+
+    private static void CreateDatabase()
+    {
+        using var connection = new SqlConnection(_connectionString);
```
🟢 **Using statements**

⭐

chrisjamiecarter commented on Flashcards.Eyad-Mostafa/FlashCards/Database/DatabaseManager.cs:
```diff
@@ -0,0 +1,341 @@
+
+            using (var command = new SqlCommand(script, connection))
+            {
+                command.Parameters.AddWithValue("@Name", stack.Name);
```
🟢 **Parameterised SQL Queries**

⭐

chrisjamiecarter commented on Flashcards.Eyad-Mostafa/README.md:
🟢 **README File**

⭐

chrisjamiecarter reviewed (changes requested):
Hey @Eyad-Mostafa 👋,

Excellent work on your **Flashcards** project submission 🎉!

I have performed a peer review. Review/ignore any comments as you wish.

***

🔴 **Study Sessions not Returned**

I have added some study sessions. But when clicking the menu option I get:

![image](https://github.com/user-attachments/assets/a7422c6c-8d0d-492c-8bb9-322fe8c06177)

Stepping through the code, you are catching an exception (👍) but not really telling the truth to the user (none found vs error getting study sessions). You write the error to console, but then immediately clear and write the none found message.

![image](https://github.com/user-attachments/assets/a0dbbbf5-e4d6-41f3-832c-8e5eecb2c1fe)

🔧 Please revisit to try understand and fix any issues marked 🔴, as they block my approval.

***

Feel free to reach out if you have any questions or if you'd like to collaborate further on any of these points 🆘 .

I will go ahead and wait for you to make your changes, commit and push (which will then notify me) 😊 !

Thanks,
@chrisjamiecarter 👍

Eyad-Mostafa commented:
Hey @chrisjamiecarter 👋,
I figured out that the problem was in creating the tables in DatabaseManager.cs file. 
the problem was that I was creating a FLOAT score and uses it as INT, so I fixed the issue and deleted the table from the SSMS and started the app again and it works now.
And I have Added Menu.PauseForUser(); for any catch block that may need to pause before continuing.

I think it's good now.
Thanks.

chrisjamiecarter reviewed (approved):
Hey @Eyad-Mostafa 👋,

Great job getting your updates pushed through 💪!

🟢 **Requirements**

⭐ You have fulfilled all of the project requirements!

I will go ahead and mark as approved, keep up the excellent work on the next projects! 😊

Best regards,
@chrisjamiecarter 👍
